utils/markdown/lines.go

```diff
@@ -3,13 +3,16 @@
 
 package markdown
 
+import "strings"
+
 type Line struct {
 	Range
 }
 
-func ParseLines(markdown string) (lines []Line) {
+func ParseLines(markdown string) []Line {
 	lineStartPosition := 0
 	isAfterCarriageReturn := false
+	lines := make([]Line, 0, strings.Count(markdown, "\n"))
```

Comment:
This is really an approximate capacity. And works for the most common case of multiple lines without carriage return.